### PR TITLE
PMM-11038 Moving to dbaas-operator

### PR DESCRIPTION
**Pull request by gen1us2k**
PMM-11038

Build: SUBMODULES-2927

- [ ] Links to other linked pull requests (optional).

### CODE REVIEW

File: api/managementpb/dbaas/pxc_clusters.proto [hunk 1]
```diff
@@ -135,6 +143,10 @@ message UpdatePXCClusterRequest {
       // Image to use. If it's the same image but with different version tag, upgrade of database cluster to version
       // in given tag is triggered. If entirely different image is given, error is returned.
       string image = 2;
+      // Configuration for PXC cluster
+      string configuration = 4;
+      // Storage Class for PXC cluster.
+      string storage_class = 5;
```
Comment (recharte) [inline api/managementpb/dbaas/pxc_clusters.proto:149]: Given that there isn't a field with index 3 shouldn't these be 3 and 4, respectively? Did you purposefully did this to be more consistent with `PXCClusterParams` above?

File: managed/Makefile [hunk 2]
```diff
@@ -11,7 +11,8 @@ help:                           ## Display this help message.
 
 # `cut` is used to remove first `v` from `git describe` output
 PMM_RELEASE_PATH ?= ../bin
-PMM_RELEASE_VERSION ?= $(shell git describe --always --dirty | cut -b2-)
+#PMM_RELEASE_VERSION ?= $(shell git describe --always --dirty | cut -b2-)
+PMM_RELEASE_VERSION ?= 2.33.0
```
Comment (recharte) [inline managed/Makefile:15]: Included in the commit by mistake?

Comment (gen1us2k) [reply managed/Makefile:15]: Yes and no. It's crucial for me at the moment and I'll remove it once I'll be ready for the review =) Thanks for pointing out!

File: api/managementpb/dbaas/pxc_clusters.proto [hunk 3]
```diff
@@ -31,6 +31,10 @@ message PXCClusterParams {
     ComputeResources compute_resources = 1;
     // Disk size in bytes.
     int64 disk_size = 2;
+    // Configuration for PXC cluster
+    string configuration = 4;
```
Comment (percona-csalguero) [inline api/managementpb/dbaas/pxc_clusters.proto:35]: Just wondering why 3 has been skipped .Will work anyway.

File: managed/services/dbaas/kubernetes/types.go [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+		resources, err := convertComputeResource(cluster.Params.Haproxy.ComputeResources)
+		if err != nil {
+			return nil, err
+		}
+		dbCluster.Spec.LoadBalancer.Image = cluster.Params.Haproxy.Image
+		dbCluster.Spec.LoadBalancer.Size = cluster.Params.ClusterSize
+		dbCluster.Spec.LoadBalancer.Resources = resources
+		dbCluster.Spec.LoadBalancer.Type = "haproxy"
+	}
+	if cluster.Params.Proxysql != nil {
```
Comment (percona-csalguero) [inline managed/services/dbaas/kubernetes/types.go:144]: Are we checking somewhere else that we have either `Haproxy` or `Proxysql`, just one of them?
I don't remember if we are doing that in another place.

File: managed/services/dbaas/kubernetes/types.go [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+			return dbCluster, fmt.Errorf("failed to recognize expose annotations for %s cluster type", clusterType)
+		}
+		dbCluster.Spec.LoadBalancer.Annotations = annotations
+		if cluster.InternetFacing && clusterType == ClusterTypeEKS {
+			dbCluster.Spec.LoadBalancer.Annotations["service.beta.kubernetes.io/aws-load-balancer-type"] = "external"
+		}
+
+	}
+	var sourceRanges []string
+	for _, sourceRange := range cluster.SourceRanges {
```
Comment (percona-csalguero) [inline managed/services/dbaas/kubernetes/types.go:171]: In the future we should validate the ranges.

File: managed/services/dbaas/kubernetes/types.go (hunk 4 above)
Comment (gen1us2k) [reply managed/services/dbaas/kubernetes/types.go:144]: No. Great catch. I'll update it soon

File: managed/services/dbaas/kubernetes/types.go (hunk 5 above)
Comment (gen1us2k) [reply managed/services/dbaas/kubernetes/types.go:171]: Agree.

File: managed/services/dbaas/kubernetes/types.go (hunk 4 above)
Comment (gen1us2k) [reply managed/services/dbaas/kubernetes/types.go:144]: Fixed it. Thanks

File: api/managementpb/dbaas/pxc_clusters.proto (hunk 1 above)
Comment (gen1us2k) [reply api/managementpb/dbaas/pxc_clusters.proto:149]: image field has index #3

File: managed/services/management/dbaas/pxc_cluster_service.go [hunk 6]
```diff
@@ -78,26 +78,24 @@ func (s PXCClustersService) GetPXCClusterCredentials(ctx context.Context, req *d
...
 	resp := dbaasv1beta1.GetPXCClusterCredentialsResponse{
 		ConnectionCredentials: &dbaasv1beta1.PXCClusterConnectionCredentials{
-			Username: cluster.Credentials.Username,
-			Password: cluster.Credentials.Password,
-			Host:     cluster.Credentials.Host,
-			Port:     cluster.Credentials.Port,
+			Username: "root",
+			Password: string(secret.Data["root"]),
+			Host:     dbCluster.Status.Host,
+			Port:     3306,
```
Comment (percona-csalguero) [inline managed/services/management/dbaas/pxc_cluster_service.go:98]: This is OK for now, but if we are allowing users to pass MySQL configuration options, like in `my.cnf`, there is a `port` option that changes the default port.

Comment (gen1us2k) [reply managed/services/management/dbaas/pxc_cluster_service.go:98]: Thanks. It's a great point. I'll create a Jira ticket for that.

File: managed/services/dbaas/kubernetes/kubernetes.go [hunk 7]
```diff
@@ -127,7 +127,7 @@ func NewEmpty() *Kubernetes {
 	}
 }
 
-// SetKubeconfig changes kubeconfig for active client
+// nges kubeconfig for active client
```
Comment (recharte) [inline managed/services/dbaas/kubernetes/kubernetes.go:130]: ?

File: managed/services/dbaas/kubernetes/kubernetes.go [hunk 8]
```diff
@@ -465,9 +465,6 @@ func (k *Kubernetes) GetAllClusterResources(ctx context.Context, clusterType Clu
 		switch clusterType {
 		case ClusterTypeUnknown:
 			return 0, 0, 0, errors.Errorf("unknown cluster type")
-		case ClusterTypeGeneric:
-			// TODO support other cluster types
-			return 0, 0, 0, nil
```
Comment (recharte) [inline managed/services/dbaas/kubernetes/kubernetes.go:434]: I suggest we keep this case but remove the action and leave a comment stating that we require it to be a no-op to execute the logic after the switch case.

Comment (gen1us2k) [reply managed/services/dbaas/kubernetes/kubernetes.go]: Done. It just continue looping

**Review by recharte — APPROVED** [review]
Good job Andrew :clap:

File: api/managementpb/dbaas/db_clusters.proto [hunk 9]
```diff
@@ -46,6 +46,8 @@ message PSMDBCluster {
   // Available database version user can upgrade cluster to, returned as an image. Image tag contains the version.
   // If it's empty, no upgrade is available.
   string available_image = 7;
+  // Is DB cluster accessible via public internet.
+  bool internet_facing = 8;
```
[the remark this answers is not in the document]
Comment (gen1us2k) [reply api/managementpb/dbaas/db_clusters.proto:50]: Nope. By default it exposes it within the VPC on EKS and to make it available you need to specify additional parameters for NLB

File: managed/services/dbaas/kubernetes/types.go [hunk 10]
```diff
@@ -0,0 +1,373 @@
...
+		resources, err := convertComputeResource(cluster.Params.Proxysql.ComputeResources)
+		if err != nil {
+			return nil, err
+		}
+		dbCluster.Spec.LoadBalancer.Size = cluster.Params.ClusterSize
+		dbCluster.Spec.LoadBalancer.Image = cluster.Params.Proxysql.Image
+		dbCluster.Spec.LoadBalancer.Resources = resources
+		dbCluster.Spec.LoadBalancer.Type = "proxysql"
+	}
+	if cluster.Expose {
```
Comment (percona-csalguero) [inline managed/services/dbaas/kubernetes/types.go:172]: What if `Expose` is true but `Haproxy` **and** `Proxysql` are both nil? Will expose work?
What if `Haproxy` and `Proxysql` both have values?

**Review by percona-csalguero — APPROVED** [review]
LGTM with a minor comment

**Pull request merged**